src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_bf_tiled.cpp

```diff
@@ -87,16 +108,45 @@ static size_t get_dynamic_quantize_group_size(const fully_connected_params& para
         }
     }
 
-    const size_t scale_group_size = params.weights.IFM().v / params.decompression_scale.Feature().v;
+    size_t scale_group_size = get_scale_group_size(params);
+    size_t zp_group_num = params.decompression_zero_point.Feature().v;
+    size_t zp_group_size = 0;
+    if (params.has_decompression_zp)
+        zp_group_size = params.weights.IFM().v / params.decompression_zero_point.Feature().v;
+
+    // Per-token dyn-quan
+    if (dynamic_quantization_group_size >= min_quantize_grp_size && is_per_token_dynamic_quantize(params)) {
+        // Validate size to fit dyn-quan group to the size of weight-scale and weight-zp
+        if ((scale_group_size % min_quantize_grp_size) == 0 && scale_group_size > min_quantize_grp_size) {
+            dynamic_quantization_group_size = scale_group_size;
+
+            // For int8 ASYM model, activation_sum should fit to weight zp
+            if (is_dyn_quan_8bit_asym(params) && params.has_decompression_zp == true &&
+                dynamic_quantization_group_size > zp_group_size && (zp_group_size % act_load_size) == 0) {
+                dynamic_quantization_group_size = zp_group_size;
+            }
+
+            GPU_DEBUG_TRACE_DETAIL << "FC dyn-quantize by per-token. Actual dyn_quan_group_size(" << dynamic_quantization_group_size
+                                    << ") : From scale_group_size (" << scale_group_size << ", zp_group_size("  << zp_group_size
+                                    << "), zp_group_num(" << zp_group_num << "), ifm_size (" << get_input_bf_size(params).second << ")" << std::endl;
```

Comment:
* Please use `LOG` instead of `TRACE_DETAIL`

Reply:
Applied